src/sagas/wallet.js

```diff
@@ -0,0 +1,541 @@
+
+  // Thread to listen for feature flags from Unleash
+  const featureFlagsThread = yield fork(listenForFeatureFlags, featureFlags);
+
+  try {
+    const serverInfo = yield call(wallet.start.bind(wallet), {
+      pinCode: pin,
+      password,
+    });
+
+    let version;
+    let networkName = network.name;
+
+    if (serverInfo) {
+      version = serverInfo.version;
+      networkName = serverInfo.network;
+    }
+
+    yield put(setServerInfo({
+      version,
+      network: networkName,
+    }));
+
+  } catch(e) {
+    if (useWalletService) {
+      // Wallet Service start wallet will fail if the status returned from
+      // the service is 'error' or if the start wallet request failed.
+      // We should fallback to the old facade by storing the flag to ignore
+      // the feature flag
+      yield call(featureFlags.ignoreWalletServiceFlag.bind(featureFlags));
+
+      // Restart the whole bundle to make sure we clear all events
+      walletHelpers.reloadElectron();
+    }
+  }
+
+  // Wallet start called, we need to show the loading addresses screen
+  routerHistory.replace('/loading_addresses');
+
+  const threads = [
+    walletListenerThread,
+    walletReadyThread,
+    featureFlagsThread
+  ];
+
+  // Wallet might be already ready at this point
+  if (!wallet.isReady()) {
+    const { error } = yield race({
+      success: take(types.WALLET_STATE_READY),
+      error: take(types.WALLET_STATE_ERROR),
+    });
+
+    if (error) {
+      yield put(startWalletFailed());
+      yield cancel(threads);
+      return;
+    }
+  }
+
+  try {
+    yield call(loadTokens);
+  } catch(e) {
+    yield put(startWalletFailed());
+    yield cancel(threads);
+    return;
+  }
+
+  routerHistory.replace('/wallet/');
+
+  // Fetch all tokens, including the ones that are not registered yet
+  const allTokens = yield call(wallet.getTokens.bind(wallet));
+
+  // Store all tokens on redux
+  yield put(loadWalletSuccess(allTokens));
+  yield put(loadingAddresses(false));
+
+  // The way the redux-saga fork model works is that if a saga has `forked`
+  // another saga (using the `fork` effect), it will remain active until all
+  // the forks are terminated. You can read more details at
+  // https://redux-saga.js.org/docs/advanced/ForkModel
+  // So, if a new START_WALLET_REQUESTED action is dispatched, we need to cleanup
+  // all attached forks (that will cause the event listeners to be cleaned).
+  yield take(types.START_WALLET_REQUESTED);
+  yield cancel(threads);
+}
+
+/**
+ * This saga will load both HTR and DEFAULT_TOKEN (if they are different)
+ * and dispatch actions to asynchronously load all registered tokens
+ */
+export function* loadTokens() {
+  const htrUid = hathorLibConstants.HATHOR_TOKEN_CONFIG.uid;
+
+  yield call(fetchTokenData, htrUid);
+
+  const registeredTokens = tokensUtils
+    .getTokens()
+    .reduce((acc, token) => {
+      // remove htr since we will always download the HTR token
+      if (token.uid === '00') {
```

Comment:
```suggestion
      if (token.uid === htrUid) {
```

Reply:
Fixed in b4d52779f64ec1563594956fe0c1e952e42ff8fe